### PR TITLE
Make Range operator to work with both Shape{} and Shape{1} inputs

### PR DESCRIPTION
**Pull request by mateusztabaka**
It's required for t2t-vit models.

### CODE REVIEW

**Comment by ilyachur** [comment]
Could you set the milestone for this activity?
Does this PR fix some ticket?

**Comment by mateusztabaka** [comment]
> Could you set the milestone for this activity?
> Does this PR fix some ticket?

There is no ticket for that.

**Review by postrational — APPROVED** [review]

File: ngraph/core/src/op/range.cpp [hunk 1]
```diff
@@ -76,12 +76,18 @@ void op::v4::Range::validate_and_infer_types()
     set_input_is_relevant_to_shape(1);
     set_input_is_relevant_to_shape(2);
 
-    NODE_VALIDATION_CHECK(
-        this, get_input_partial_shape(0).compatible(Shape{}), "'start' input is not a scalar");
-    NODE_VALIDATION_CHECK(
-        this, get_input_partial_shape(1).compatible(Shape{}), "'stop' input is not a scalar");
-    NODE_VALIDATION_CHECK(
-        this, get_input_partial_shape(2).compatible(Shape{}), "'step' input is not a scalar");
+    NODE_VALIDATION_CHECK(this,
```
Comment (lazarevevgeny) [inline ngraph/core/src/op/range.cpp:79]: These changes contradicts with the spec. So they are not correct.

Also, take a look at https://github.com/openvinotoolkit/openvino/pull/6008 and https://github.com/openvinotoolkit/openvino/pull/6017 which may have fixed the root cause why the Range got a tensor of shape [1] as input.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Comment by mateusztabaka** [comment]
On latest master I'm not able to reproduce the issue. Closing.

**Pull request closed without merging**